xbout/tests/test_load.py

```diff
@@ -947,16 +988,23 @@ def test_combine_along_xy(self, tmpdir_factory, bout_xyt_example_files):
         )
         expected = concat([line1, line2, line3], dim="y", data_vars="minimal")
         expected = expected.set_coords("t_array").rename(t_array="t")
+        vars_to_drop = METADATA_VARS + _BOUT_PER_PROC_VARIABLES
+        if bout_v5:
+            vars_to_drop.remove("dz")
```

Comment:
Why is this necessary? Shouldn't `actual` contain `"dz"` since it was added at l.659? I'm obviously missing something, because the tests pass at the moment!

Reply:
Er, I forget exactly why, but this was causing me issues. I'll double check to see if this is still necessary

Reply:
Ah because these are scalars/metadata to drop, and with BOUT++ v5, `dz` doesn't appear there so if we don't remove it then it only appears on one side of the comparison.

If I can work out how to move `dz` out of the metadata into the "real" DataArray, then both v4 and v5 should be able to handle it correctly without special handling here

Reply:
I think I know where to move `dz` out of metadata - probably easier if I just make the change, unless you've done it already @ZedThree?

Reply:
Yes please, I have just been trying and failing to do it :smile:

Reply:
Sorry, we could probably do with some developer docs, but I doubt I'll ever have time to write them...